src/login/user-runtime-dir.c

```diff
@@ -67,7 +67,7 @@ static int user_mkdir_runtime_path(
         if (r < 0)
                 return log_error_errno(r, "Failed to create /run/user: %m");
 
-        if (path_is_mount_point(runtime_path, NULL, 0) >= 0)
+        if (path_is_mount_point(runtime_path) > 0)
```

Comment:
Could you split out the fix at first? Then we can easily backport the fix to stable repository if necessary.